### ISSUE TITLE
[Feat] ParallelModuleQueue (python multiprocessing): don't wait for entire block to finish before pulling new processes 

### ISSUE THREAD

**Issue by griembauer** [OSGeo/grass#2478]
The option to run GRASS modules in parallel (in python) is implemented via the [ParallelModuleQueue](https://github.com/OSGeo/grass/blob/d0b904f6e9077ee5f09f80e8ddbc9e7bb5b2c050/python/grass/pygrass/modules/interface/module.py#L37) class. The standard way (?) is to define a processing queue via an `nprocs` parameter, add GRASS modules to be executed in parallel via the `put()` method and finally start the parallel processing using the `wait()` method. 
The way it is implemented now, the queue seems to run a number of processes defined by `nprocs` and waits for **all** processes to finish before starting the next "block" of processes. This means that the longest process determines the duration of an entire processing "block". 
Ideally, free slots could be filled directly with pending processes from the queue instead.

**Comment by petrasovaa** [comment]
I agree that is a problem, which is partially the reason I usually just use standard Python multiprocessing.Pool methods (like `map_async`) with `run_command`. Just curious, do you prefer ParallelModuleQueue for some specific reason?

**Comment by griembauer** [comment]
No, not at all, I am just used to using it since it is the _pygrass_ way ;)
